sympy/polys/rings.py

```diff
@@ -446,19 +451,19 @@ def term_new(self, monom, coeff) -> PolyElement[Er]:
         return poly
 
     # Polynomial creation from various formats
-    def from_dict(self, element, orig_domain=None) -> PolyElement[Er]:
+    def from_dict(self, element: dict[Mon, int | Er | Expr], orig_domain: Domain[Er] | None =None) -> PolyElement[Er]:
```

Comment:
The dict type is "invariant" in the value type because it is mutable. In cases like this it is better to use `Mapping` rather than dict.